### PR TITLE
Clang format linter

### PR DESCRIPTION
**Pull request by LucasGandel**
@SimonRit This should run clang-format on PR and make the pipeline fail if the style is inconsistent. For branches pushed on the repo (not fork), style can automatically be fixed by adding the label action:ApplyClangFormat

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-*.hxx            our-c-style
-*.txx            our-c-style
-*.txt            whitespace=tab-in-indent,no-lf-at-eof
-*.cmake          whitespace=tab-in-indent,no-lf-at-eof
+include/*.h*                            our-c-style
+src/*.c*                                our-c-style
+applications/**/*.h                     our-c-style
+applications/**/*.cxx                   our-c-style
+applications/**/*.hxx                   our-c-style
+examples/*.cxx                          our-c-style
```
Comment (SimonRit) [inline .gitattributes]: Shouldn't this be `examples/**/*.cxx`?

**Review by SimonRit — CHANGES_REQUESTED** [review]
Ok thanks. Can you rebase your 3 commits on the updated #329? I have included the cmake/*cxx files, maybe you want to do the same. There is also one comment to address.
Locally, will this automatically apply the style as is done on a commit in the ITK repository?

File: .gitattributes (hunk 1 above)
Comment (LucasGandel) [reply .gitattributes]: Nice catch! Thanks!

**Comment by LucasGandel** [comment]
@SimonRit I made the changes you requested, thanks.
The linter will run automatically on each PR as soon as this is merged. 
For automatically fixing the style, you must add the "action:ApplyClangFormat" label to your PR. But this works only for branches pushed to your repository (not forks).
I am still working on the hooks to fix style locally but this is an additional check.

**Pull request merged**